### ISSUE TITLE
Partition Metrics based on Enum attributes fail

### ISSUE THREAD

**Issue by adamthehutt** [laravel/nova-issues#3705]
- Laravel Version: 8.78.1
- Nova Version: 3.30.0
- PHP Version: 8.1.1

### Description:
PHP 8.1 introduced Enums, and Laravel now supports them as an attribute cast. However, in Nova a Partition Metric derived from an Enum attribute results in an "Invalid offset type" error in Metrics/Partition.php:142.

The offending method is:
```php
protected function formatAggregateResult($result, $groupBy)
{
    $key = $result->{last(explode('.', $groupBy))};
    return [$key => $result->aggregate];
}
```

I believe the code is trying to use the Enum itself as the array key, which is not valid. 

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create an Eloquent model with an Enum as one of the attribute casts. 
2. Create a Partition Metric that counts/groups by that attribute.

**Comment by crynobone** [comment]
Please provide reproducing code to debig the issue.

**Comment by adamthehutt** [comment]
Hopefully this is enough to illustrate the issue:

```php

enum MakeOfCar
{
    case Toyota = 'T';
    case Nissan = 'N';
    case Honda = 'H';
}

class Car extends Model 
{
    protected $casts = [
        'make' => MakeOfCar::class
    ];
}

class CarsByModel extends Partition
{
    public function calculate(NovaRequest $request)
    {
        return $this->count($request, Car::class, 'make');
    }
}

```

The problem is that an Enum can't be used as an array key, so the code in formatAggregateResult() throws a TypeError.

**Comment by adamthehutt** [comment]
This issue has not been resolved, so please don't close it.

**Comment by davidhemphill** [comment]
👋 Hi there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by crynobone** [comment]
Will look into adding the support soon.

**Comment by crynobone** [comment]
```php
    protected function formatAggregateResult($result, $groupBy)
    {
        $key = $result->{last(explode('.', $groupBy))};
        $key = with($result->{last(explode('.', $groupBy))}, function ($key) {
            if ($key instanceof \BackedEnum) {
                return $key->value;
            }

            return $key;
        });

        return [$key => $result->aggregate];
```

Can you test this?

**Comment by adamthehutt** [comment]
@crynobone Yes, that works. I don't know if you'd want something that supports attribute casts more generically, but if that's not needed then this solves the problem.

**Comment by crynobone** [comment]
Fixed in v3.31.0: https://nova.laravel.com/releases/3.31.0
